app/main.py

```diff
@@ -494,9 +494,22 @@ def initialize(self, request, response):
 
     def serve(self):
         request, response, env = self.request, self.response, self.env
+
+        # If the Person Finder instance has not been initialized yet,
+        # prepend to any served page a warning and a link to initialize
+        # the datastore.
+        if 0 == len(env.repo_options) and env.action != 'setup_datastore':
+            get_vars = lambda: {'env': env}
+            content = resources.get_rendered('setup_datastore.html', env.lang,
+                    (env.repo, env.charset), get_vars)
+            response.out.write(content)
+
         if not env.action and not env.repo:
             # Redirect to the default home page.
             self.redirect(env.global_url + '/' + HOME_ACTION)
+        elif env.action == "setup_datastore":
```

Comment:
Having two end points to set up datastore looks weird. (One defined here, another defined in admin.py.) How about linking to the end point defined in admin.py from setup_datastore.html? Anyway it should be POST, not GET.